File: python_translation/compute_GFT_noQ.py
def compute_GFT_noQ(Adj, A, idx_closest=None):
    """
    Compute the Graph Fourier Transform (GFT) without using the quality matrix.

    Parameters:
        Adj (numpy.ndarray): Adjacency matrix of the graph.
        A (numpy.ndarray): Attribute matrix.
        idx_closest (numpy.ndarray or None): Index of the closest points (optional).

    Returns:
        numpy.ndarray: Graph Fourier Transform.
        numpy.ndarray: Eigenvalues of the Laplacian matrix (sorted in ascending order).
        numpy.ndarray: Transformed attribute matrix.
    """
    if idx_closest is not None:
        L = w2l(Adj, idx_closest)
    else:
        L = w2l(Adj)

    GFT, D = np.linalg.eig(L)
    idxSorted = np.argsort(np.diag(D))

    GFT = GFT[:, idxSorted]
    GFT[:, 0] = np.abs(GFT[:, 0])
    GFT = GFT.T
    Gfreq = np.abs(np.diag(D))
    Gfreq[0] = np.abs(Gfreq[0])

    Ahat = np.dot(GFT, A)

    return GFT, Gfreq, Ahat

File: matlab_code/compute_GFT_noQ.m
function [ GFT, Gfreq, Ahat ] = compute_GFT_noQ(Adj, A, idx_closest)

if(nargin > 2)
    L = w2l(Adj,idx_closest);
else
    L = w2l(Adj);
end
[GFT,D] = eig(L);       % GFT is right eigenvectors in columns D is diagonal matrix with eigenvalues
[Gfreq, idxSorted ] = sort( diag(D), 'ascend' );
GFT = GFT(:, idxSorted);
GFT(:,1)=abs(GFT(:,1));
GFT = GFT';
Gfreq(1) = abs(Gfreq(1));
Ahat = GFT * A;
end